transformers/uws/server.js

```diff
@@ -46,7 +46,12 @@ module.exports = function server() {
       });
     });
 
-    this.service.upgrade(soc._handle.fd, req.headers['sec-websocket-key'], soc.ssl ? soc.ssl._external : null);
+    var ticket = this.service.transfer(soc._handle.fd, soc.ssl ? soc.ssl._external : null);
+    soc.on('close', function(error) {
+      if (!error) {
```

Comment:
https://nodejs.org/api/net.html#net_event_close_1

'close' is called with a boolean true if a transmission error occurred (in these cases we do not want to upgrade). Otherwise, destroy calls close with false.